flytekit/core/base_task.py

```diff
@@ -203,12 +203,13 @@ def __init__(
         task_type_version=0,
         security_ctx: Optional[SecurityContext] = None,
         docs: Optional[Documentation] = None,
+        timeout: Optional[Union[datetime.timedelta, int]] = None,
```

Comment:
Would it make more sense to expose `metadata` to the `PythonTask` and have `BaseSensor` initialize the metadata w/ `TaskMetadata(timeout=timeout)`?

Reply:
yes it will make more sense.